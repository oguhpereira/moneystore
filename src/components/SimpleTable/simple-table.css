.table-content {
	margin-top: 35px;
}

.simple-table {
	width: 100%;

	text-align: center;

	border-collapse: collapse;
}

.simple-table .thead {
	margin-bottom: 10px;

	background-color: var(--gray-light);
}

.simple-table .tbody {
	background-color: white;
}

.simple-table .th {
	padding: 15px;

	color: #979797;

	font-weight: 400;
}

.simple-table .td {
	padding: 15px;
}
