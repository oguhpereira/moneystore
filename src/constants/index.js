export const API_MERCADO_BITCOIN = 'https://www.mercadobitcoin.net/api';
export const API_BANCO_CENTRAL =
	'https://olinda.bcb.gov.br/olinda/servico/PTAX/versao/v1/odata';

export const INVALID_PARAM = 'INVALID_PARAM';
export const PARAM_VERYSHORT = 'PARAM_VERYSHORT';
export const INVALID_QUOTE = 'INVALID_QUOTE';
export const REQUEST_SUCCESSFUL = 'REQUEST_SUCCESSFUL';
export const REQUEST_FAILID = 'request failed';
export const INSUFFICIENT_MONEY = 'INSUFFICIENT_MONEY';
export const INSUFFICIENT_QUANTITY = 'INSUFFICIENT_QUANTITY';
export const INVALID_TOKEN = 'INVALID_TOKEN';
export const FORM_VALIDATION = 'FORM_VALIDATION';
export const RENDER_VALIDATION = 'RENDER_VALIDATION';
