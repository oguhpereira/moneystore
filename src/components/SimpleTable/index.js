/**
 * @author oguhpereira
 * Simple table component
 */
import React from 'react';
import './simple-table.css';

/**
 * @function TableRow
 * @param {Array} data
 * @param {Array} keys
 * @returns {JSX}
 */
function TableRow({ data, keys }) {
	return (
		<tr>
			{keys.map((td, indextd) => (
				<td className="td" key={indextd}>
					{data[td]}
				</td>
			))}
		</tr>
	);
}

/**
 * @function SimpleTable
 * @param {String} title
 * @param {Array} head
 * @param {Array} content
 * @param {Array} keys
 * @returns {JSX}
 */
export function SimpleTable({ title, head, content = [], keys }) {
	if (content.length > 0) {
		return (
			<div className="table-content">
				{title && <h2 className="subtitle">{title}</h2>}
				<table className="simple-table">
					<thead className="thead">
						<tr>
							{head.map((nameTH, index) => (
								<th className="th" key={index}>
									{nameTH}
								</th>
							))}
						</tr>
					</thead>
					<tbody className="tbody">
						{content.map((tr, index) => {
							return <TableRow key={index} data={tr} keys={keys} />;
						})}
					</tbody>
				</table>
			</div>
		);
	}
	return <div />;
}
