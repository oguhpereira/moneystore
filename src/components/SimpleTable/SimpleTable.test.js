import React from 'react';
import { shallow } from 'enzyme';

import SimpleTable from './index';
import { RENDER_VALIDATION } from '../../constants';

const setUp = (props) => shallow(<SimpleTable {...props} />);

describe('SimpleTable Component', () => {
	let component;
	let mockProps = {};
	beforeEach(() => {
		mockProps = {
			title: 'Simple Table',
			head: ['Name', 'Price'],
			keys: ['name', 'price'],
			content: [{ name: 'Test1', price: 1000 }],
		};

		component = setUp(mockProps);
	});

	it(RENDER_VALIDATION, () => {
		expect(component.debug()).toMatchSnapshot();
	});
});
