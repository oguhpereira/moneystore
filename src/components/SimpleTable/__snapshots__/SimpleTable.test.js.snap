// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleTable Component RENDER_VALIDATION 1`] = `
"<div className=\\"table-content\\">
  <h2 className=\\"subtitle\\">
    Simple Table
  </h2>
  <table className=\\"simple-table\\">
    <thead className=\\"thead\\">
      <tr>
        <th className=\\"th\\">
          Name
        </th>
        <th className=\\"th\\">
          Price
        </th>
      </tr>
    </thead>
    <tbody className=\\"tbody\\">
      <TableRow data={{...}} keys={{...}} />
    </tbody>
  </table>
</div>"
`;
